### ISSUE TITLE
custom certificate used for all domains instead of CN specified domain only

### ISSUE THREAD

**Issue by markraemer** [mitmproxy/mitmproxy#1461]
##### Steps to reproduce the problem:
following the guide for custom certificates at http://docs.mitmproxy.org/en/stable/certinstall.html
1.  use openssl as described to create a certificate for a specific domain
2. start mitmproxy with --cert=cert.pem
3. mitmproxy intercepts all https connections using that certificate

##### What is the expected behavior?
mitmproxy would use that certificate just for the domain specified in the certificate's CN

##### What went wrong?
mitmproxy uses this very same certificate for all https connections, I think defeating the CN mapping

##### Any other comments? What have you tried so far?
verified that this certificate is used instead of mitmproxy generated and self signed for another domain then specified

---

Mitmproxy Version: v0.17.1
Operating System: Kali Linux 2.0

**Comment by mhils** [comment]
Whoops - looks like the docs are a bit flawed there.

```
λ mitmproxy --help

  --cert SPEC           Add an SSL certificate. SPEC is of the form
                        "[domain=]path". The domain may include a wildcard,
                        and is equal to "*" if not specified. The file at path
                        is a certificate in PEM format. If a private key is
                        included in the PEM, it is used, else the default key
                        in the conf dir is used. The PEM file should contain
                        the full certificate chain, with the leaf certificate
                        as the first entry. Can be passed multiple times.
```

I'd be super happy to accept a PR that fixes the `--cert` invocation in the docs! :smiley:

**Comment by mhils** [comment]
Fixed by #1609! :smiley:
